pgex/examples/buttons/button.py

```diff
@@ -36,19 +37,15 @@ def draw(self, surface: pygame.Surface) -> None:
             self._text_pos,
         )
 
-    def update(self) -> None:
-        """Must be called every frame."""
-        if (
-            self.rect.collidepoint(pygame.mouse.get_pos())
-            and pygame.mouse.get_pressed()[0]
-            and not self._is_pressed
-        ):
-            self.func()
-            self._is_pressed = True
-
-        elif (
-            self.rect.collidepoint(pygame.mouse.get_pos())
-            and not pygame.mouse.get_pressed()[0]
-            and self._is_pressed
-        ):
+    def handle_events(self, event: pygame.event.Event) -> None:
+        """Put this inside of your event loop"""
+        if event.type == pygame.MOUSEBUTTONDOWN:
+            if event.button == 1 and self.rect.collidepoint(event.pos):
+                self._is_pressed = True
+            else:
+                self._is_pressed = False
```

Comment:
This whole condition block could be simplified like this:

```py
self.is_pressed = event.button == 1 and self.rect.collidepoint(event.pos)
```

Reply:
Nice catch, I guess somebody could make a PR if they want to change that, but we should also discuss how beginner friendly this would be (I mean, this already uses OOP and stuff...) although I'd be down for that change.

Reply:
Fair point, direct assignment of the result of a predicate expression to an identifier might be too much for a beginner.
But in the other hand, is it better to show them non-optimized solutions?

Reply:
I'd think that this really doesn't hinder performance noticeably, so it's not that big of an issue anyways.